daemon/daemon_unix.go

```diff
@@ -375,6 +390,13 @@ func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.
 		warnings = append(warnings, "IPv4 forwarding is disabled. Networking will not work.")
 		logrus.Warnf("IPv4 forwarding is disabled. Networking will not work")
 	}
+	if hostConfig.CgroupParent != "" && daemon.usingSystemd() {
+		// CgroupParent for systemd cgroup should be named as "xxx.slice"
+		length := len(hostConfig.CgroupParent)
+		if length <= 6 || hostConfig.CgroupParent[length-6:] != ".slice" {
```

Comment:
Basically, using `--cgroup-parent=some.slice` will do weird things with systemd if your system is under load (`systemd` has the right to move the Docker process to the `system.slice` when it feels like it, so you end up with not joining the slice you might've thought you were joining). Really, it just means that we should only mandate slice names to be valid if it's an absolute path IMO.